tests/seeding/test_dataset_deletes.py

```diff
@@ -0,0 +1,255 @@
+
+from typing import Iterator
+from uuid import UUID
+
+from citrine.resources.dataset import Dataset
+from citrine.resources.data_concepts import DataConcepts
+from citrine.resources.delete import DELETE_SERVICE_MAX
+from citrine.resources.property_template import PropertyTemplateCollection, PropertyTemplate
+from citrine.resources.condition_template import ConditionTemplateCollection, ConditionTemplate
+from citrine.resources.parameter_template import ParameterTemplateCollection, ParameterTemplate
+from citrine.resources.process_template import ProcessTemplateCollection, ProcessTemplate
+from citrine.resources.material_template import MaterialTemplateCollection, MaterialTemplate
+from citrine.resources.measurement_template import MeasurementTemplateCollection, MeasurementTemplate
+from citrine.resources.process_run import ProcessRunCollection, ProcessRun
+from citrine.resources.material_run import MaterialRunCollection, MaterialRun
+from citrine.resources.measurement_run import MeasurementRunCollection, MeasurementRun
+from citrine.resources.ingredient_run import IngredientRunCollection, IngredientRun
+from citrine.resources.process_spec import ProcessSpecCollection, ProcessSpec
+from citrine.resources.material_spec import MaterialSpecCollection, MaterialSpec
+from citrine.resources.measurement_spec import MeasurementSpecCollection, MeasurementSpec
+from citrine.resources.ingredient_spec import IngredientSpecCollection, IngredientSpec
+from citrine.seeding.dataset_deletes import wipe_dataset
+from tests.utils.session import FakeSession
+
+from gemd.demo.cake import make_cake, make_cake_templates, make_cake_spec, make_cake_templates
+from gemd.util import flatten
+
+import math
+import pytest
+
+
+@pytest.fixture
+def fake_dataset():
+    """Fake dataset that serves GEMD objects via collections' list_all methods."""
+
+    cake_spec = make_cake_spec()
+    cake_tmpls = make_cake_templates()
+    gemds = [DataConcepts.build(gemd) for i in range(8) for gemd in flatten(make_cake(tmpl=cake_tmpls, cake_spec=cake_spec), "dummy")]
+
+
+    class FakePropertyTemplateCollection(PropertyTemplateCollection):
+        def __init__(self):
+            pass
+
+        def list_all(self, forward: bool = True, per_page: int = 100) -> Iterator[PropertyTemplate]:
+            return iter([x for x in gemds if isinstance(x, PropertyTemplate)])
```

Comment:
If you adopt the changes, above each of these need to read
```suggestion
        def list(self,
                page: Optional[int] = None,
                per_page: Optional[int] = 100) -> List[PropertyTemplate]:
            return [x for x in gemds if isinstance(x, PropertyTemplate)]
```

Reply:
DE prefers using `list_all`. We usually never use `list`